File: src/main/java/com/yupi/airouter/plugin/PluginResult.java
/**
 * 插件执行结果
 *
 * @author <a href="https://codefather.cn">编程导航学习圈</a>
 */
package com.yupi.airouter.plugin;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PluginResult {

    /**
     * 是否执行成功
     */
    private boolean success;

    /**
     * 结果内容（可以直接添加到对话上下文）
     */
    private String content;

    /**
     * 错误信息
     */
    private String errorMessage;

    /**
     * 执行耗时（毫秒）
     */
    private long duration;

    /**
     * 额外数据
     */
    private Map<String, Object> data;

    /**
     * 创建成功结果
     */
    public static PluginResult success(String content) {
        return PluginResult.builder()
                .success(true)
                .content(content)
                .build();
    }

    /**
     * 创建成功结果（带数据）
     */
    public static PluginResult success(String content, Map<String, Object> data) {
        return PluginResult.builder()
                .success(true)
                .content(content)
                .data(data)
                .build();
    }

    /**
     * 创建失败结果
     */
    public static PluginResult fail(String errorMessage) {
        return PluginResult.builder()
                .success(false)
                .errorMessage(errorMessage)
                .build();
    }
}

File: python-backend/app/plugin/plugin_result.py
"""Plugin execution result."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any


@dataclass(slots=True)
class PluginResult:
    success: bool
    content: str | None = None
    error_message: str | None = None
    duration: int = 0
    data: dict[str, Any] = field(default_factory=dict)

    @staticmethod
    def success_result(content: str, data: dict[str, Any] | None = None) -> "PluginResult":
        return PluginResult(success=True, content=content, data=data or {})

    @staticmethod
    def fail(error_message: str) -> "PluginResult":
        return PluginResult(success=False, error_message=error_message)
